load.php

```diff
@@ -253,9 +263,56 @@ function perflab_load_active_and_valid_modules() {
 		require_once PERFLAB_PLUGIN_DIR_PATH . 'modules/' . $module . '/load.php';
 	}
 }
-
 perflab_load_active_and_valid_modules();
 
+/**
+ * Places the Performance Lab's object cache drop-in in the drop-ins folder.
+ *
+ * This only runs in WP Admin to not have any potential performance impact on
+ * the frontend.
+ *
+ * This function will short-circuit if the constant
+ * 'PERFLAB_DISABLE_OBJECT_CACHE_DROPIN' is set as true.
+ *
+ * @since n.e.x.t
+ */
+function perflab_maybe_set_object_cache_dropin() {
+	global $wp_filesystem;
+
+	// Bail if disabled via constant.
+	if ( defined( 'PERFLAB_DISABLE_OBJECT_CACHE_DROPIN' ) && PERFLAB_DISABLE_OBJECT_CACHE_DROPIN ) {
+		return;
+	}
+
+	// Bail if already placed.
+	if ( PERFLAB_OBJECT_CACHE_DROPIN_VERSION ) {
+		return;
+	}
+
+	// Bail if already attempted before timeout has been completed.
+	// This is present in case placing the file fails for some reason, to avoid
+	// excessively retrying to place it on every request.
+	$timeout = get_transient( 'perflab_set_object_cache_dropin' );
+	if ( false !== $timeout ) {
+		return;
+	}
+
+	if ( $wp_filesystem || WP_Filesystem() ) {
+		// If there is an actual object-cache.php file, rename it.
+		// The Performance Lab object-cache.php will still load it, so the
+		// behavior does not change.
+		if ( $wp_filesystem->exists( WP_CONTENT_DIR . '/object-cache.php' ) ) {
+			$wp_filesystem->move( WP_CONTENT_DIR . '/object-cache.php', WP_CONTENT_DIR . '/object-cache-orig.php' );
```

Comment:
What happens when the user deactivates the plugin? Do we revert this change and rename `object-cache-orig.php` to `object-cache.php` otherwise old file not works?

Reply:
@mukeshpanchal27 Great catch! I've addressed this in https://github.com/WordPress/performance/pull/553/commits/6f2f1c59cbb294c4871bf14ac4ba3912176796b5.

Even if the file is not removed, it wouldn't cause problems since its code will still include the original drop-in, and only load the Performance Lab code if it is present. However I agree with you that it's a best practice to remove our custom drop-in on plugin deactivation.